README.md

```diff
@@ -1,135 +1,268 @@
+
+Results from the command are expected to be streamed in JSON line
+format, and the FAW can be optionally configured to capture and render
+custom HTML files generated by the plugin. This flexibility allows
+decision plugins to introduce new UI elements and actions within the
+FAW. FAW's root cause analysis are implemented as decision plugins.
+
+### Pipeline Plugins
+
+The FAW supports complex plugins with multiple stages and the capability
+to generate new data-driven parsers based on corpus inputs.
+
+TODO: Point to plugin tutorial
+
+### Transforming Inputs and Universal Output Parsers
+
+The FAW allows for input transformers that dynamically modify inputs
+before processing them. These transformers take a file from the corpus
+as input run external commands that produce transformed content on the
+standard output. Transformed inputs are treated as transient and not
+permanently stored. The SDF distribution, for example, use input
+transformers to generate a large corpus of SDF files from very simple
+seed input.
+
+The FAW also supports universal parser parsers, which operate on the
+output of all configured parsers, generating additional features. These
+parsers run external commands that accept specially formatted streams
+and emit JSON dictionaries of new features. Universal parser parsers
+unify format-specific considerations across all parsers
+
+## Root Cause Analysis
+
+TODO: Ask!
+
+## Rendering differentials
+
+The FAW includes a rendering differential detection algorithm designed
+for file formats that generate on-screen images like PDFs. Unlike
+existing algorithms that often report false positives, this algorithm
+reduces such errors and accurately identifies true differentials,
+enabling the identification of root causes for the differences.
+
+## PolyFile / PolyTracker Integration
+
+Two essential tools included in most FAW distributions are PolyFile and
+PolyTracker. PolyFile serves as a unified file format tool with parsers
+for over 263 MIME file types. It detects polyglots (valid instances of
+multiple formats) and provides a file detail view for inspecting parsed
+structures or byte regions. PolyTracker is an instrumentation toolkit
+that enhances existing parsers with taint-tracking functionality,
+including control flow tracking algorithms for grammar inference and the
+identification of "blind spots" that do not affect a parser's
+interpretation of bytes in a file, even when mutated.
+
+## Grammar Inference / Pipeline Plugins
+
+Parsers like those learned through machine learning require additional
+support to handle the training process. The FAW includes a pipelining
+plugin system, which allows long running actions to be divided in to a
+acyclic graph of resumable tasks. Tasks are processed according to the
+graph with restarting of crashed tasks. Once tasks are completed,
+parsers can be defined for analysis sets, file detail views, and
+decision plugins. RL-GRIT has been integrated in to FAW using pipeline
```

Comment:
Probably worth a link/explanation for context

Reply:
Plus 1 to this. RL-GRIT isn't referenced prior to this point. We could link one of the LangSec papers on the tool?

Reply:
I have linked to the RL-GRIT paper. (See commit [c7d9153](https://github.com/GaloisInc/FAW/pull/94/commits/c7d9153ac4aacdbffc60c38bb19d27d9908c8ad9)). Should we do more?

Reply:
At some point, but I think for now this is fine